### PR TITLE
Add keyboard shortcut styling to "/" in default tip in the inserter modal

### LINKED ISSUE #17111: Add a "keyboard shortcut" treatment to the default tip in the inserter modal
Followup from #16813.

The default tip in the inserter modal currently says: 

Screen Shot 2019-08-20 at 11 41 47 AM

To make the `/` part of this message more clear, we should treat the `/` similar to how Gutenberg displays [keyboard shortcuts](https://cloudup.com/cjl7XbzJdFv) in the shortcuts modal: 

code

### PR DESCRIPTION
**Pull request by enriquesanchez**
Closes #17111 

This PR adds keyboard shortcut styling to the "/" in the default tip in the inserter modal.

**Before:**
63377650-7e035300-c35e-11e9-9af5-6383f554b1fc

**After:**
Screen Shot 2019-08-27 at 3 11 01 PM

### CODE REVIEW

File: packages/block-editor/src/components/inserter/menu.js
Comment (talldan) [inline packages/block-editor/src/components/inserter/menu.js]: The class name applied here is from another component, which isn't really something encouraged as it can lead to maintainability issues (i.e. in the future someone changes the other component and doesn't realise this one will be affected).

I think there are three options:
- Introduce a generic `<Keyboard>` react component in the `@wordpress/components` package that has this styling and can be use across the editor. There would have to be some thought about whether this is something wanted in that package.
- Introduce some global styling for the `kbd` element or a mixin. Not sure about this option as I think it's nice to keep the global styling to a minimum and promote using components instead.
- Duplicate some of the styles in tip component.

I think the third one might be preferable. As far as I know there are only two places in the codebase that have `kbd` styles. When there are three or four it might be time to consider refactoring to a component.

File: packages/block-editor/src/components/inserter/menu.js
Comment (talldan) [inline packages/block-editor/src/components/inserter/menu.js]: I'm not completely sure of the best way to internationalise a string that has an HTML tag in it. Maybe using `sprintf`. I couldn't see any examples in the codebase. I'll ask for some feedback on that.

**Review by talldan — CHANGES_REQUESTED** [review]
Thanks for working on this @enriquesanchez, this is something that makes complete sense and I'd really like to see merged!

Had a couple of comments, hopefully no major blockers if we can find a solution for the internationalisation.

File: packages/block-editor/src/components/inserter/menu.js
Comment (talldan) [reply packages/block-editor/src/components/inserter/menu.js]: Feedback on slack is here: https://wordpress.slack.com/archives/C02QB2JS7/p1566976018008100 (requires an account)

Seems as though splitting the string isn't a good idea, but there's no solution yet to internationalising text that has an html element in it. https://github.com/WordPress/gutenberg/pull/16374 is a PR that attempts to address it.

Comment (gziolo) [reply packages/block-editor/src/components/inserter/menu.js]: Yes. it's very unfortunate that we can't use HTML tags inside translations as of today :(

@nerrad - how far did you get with your implementation for Interpolate component?

Comment (talldan) [reply packages/block-editor/src/components/inserter/menu.js]: Unfortunately I think it is a blocker. 😞 

Unless there's a workaround I think it'll hold up the PR.

**Comment by enriquesanchez** [comment]
Closing because this PR is blocked until #16374 is merged. 
Thanks everyone for the reviews! 👍

File: packages/block-editor/src/components/inserter/menu.js
Comment (nerrad) [reply packages/block-editor/src/components/inserter/menu.js]: Sorry for the delay here:

> how far did you get with your implementation for Interpolate component?

In the pull I did, the more verbose `createInterpolateElement` can be extracted from that pull and released as is.  It's usable and will solve the problem.  I'm mostly waiting on whether that's a path we want to take initially.

However, the more intuitive jsx-like option with a `<Translate>` or `<Interpolate/>` component is not available yet (maybe could be done on a separate pull?).

**Comment by gziolo** [comment]
We discussed this PR during the weekly Core JS chat today. @nerrad will work on `createInterpolateComponent` helper function which should help us unblock this PR. Related comment: 
https://github.com/WordPress/gutenberg/pull/16374#issuecomment-527491049

File: packages/block-editor/src/components/inserter/menu.js
Comment (gziolo) [reply packages/block-editor/src/components/inserter/menu.js]: I opened https://github.com/WordPress/gutenberg/pull/18623 to add the same change but using the newly introduced API method.

**Comment by talldan** [comment]
Looks like this is now unblocked with the `__experimentalCreateInterpolateElement` now merged to `master`.

**Comment by talldan** [comment]
@nerrad Would the correct usage be something like:
```jsx
createInterpolateElement(
    __( 'While writing, you can press <kbd>/</kbd> to quickly insert new blocks' ),
    { kbd: <kbd /> }
);
```

**Comment by talldan** [comment]
Oh, just noticed this has already been handled in a separate PR, https://github.com/WordPress/gutenberg/pull/18623. Ignore my above comments!

**Comment by gziolo** [comment]
😄 

There is an ongoing discussion whether `<kbd>/<kbd>` should be replaced with something else in the translations. I like the idea of having something like:

```js
createInterpolateElement(
    __( 'While writing, you can press <SlashKey /> to quickly insert new blocks' ),
    { SlashKey: <kbd>/</kbd> }
);```

**Pull request closed without merging**
